crates/rpc/rpc/src/eth/pubsub.rs

```
@@ -106,6 +109,58 @@ async fn handle_accepted<Client, Pool, Events>(
+                let initial_sync_status = pubsub.client.is_syncing();
+
+                let initial_sub_res = if initial_sync_status {
+                    EthSubscriptionResult::SyncState(PubSubSyncStatus::Detailed(
+                        SyncStatusMetadata {
+                            syncing: true,
+                            starting_block: 0,
+                            current_block: pubsub.client.chain_info().unwrap().best_number,
+                            highest_block: None,
+                        },
+                    ))
+                } else {
+                    EthSubscriptionResult::SyncState(PubSubSyncStatus::Simple(false))
+                };
+
+                match accepted_sink.send(&initial_sub_res) {
+                    Ok(_) => {
+                        // The message was successfully sent
+                    }
+                    Err(_) => {
+                        // An error occurred while sending the message
+                    }
+                }
+
+                while let Some(_) = block_subscription.recv().await {
+                    let sub_res = if pubsub.client.is_syncing() {
+                        EthSubscriptionResult::SyncState(PubSubSyncStatus::Detailed(
+                            SyncStatusMetadata {
+                                syncing: true,
+                                starting_block: 0,
+                                current_block: pubsub.client.chain_info().unwrap().best_number,
+                                highest_block: None,
+                            },
+                        ))
+                    } else {
+                        EthSubscriptionResult::SyncState(PubSubSyncStatus::Simple(false))
+                    };
+
+                    if pubsub.client.is_syncing() != initial_sync_status {
+                        match accepted_sink.send(&sub_res) {
+                            Ok(_) => {
```

can ignore the error for now.